examples/Processor.hs

```diff
@@ -64,10 +64,19 @@ type InstructionAddress = Value
 data State = State { registers :: RegisterBank
                    , memory    :: Memory
                    , pc        :: InstructionAddress
-                   , flags     :: Flags }
+                   , flags     :: Flags
+                   , logs      :: [String] }
```

Comment:
Let's make the `log` a bit more type-safe, as below, to be able to distinguish reads from writes safely:

https://github.com/snowleopard/build/blob/8ea196802910337643eaafae4a0a902b8ea29edf/src/Build/Task/Opaque.hs#L103